src/rewriter.fs

```diff
@@ -537,18 +537,13 @@ let private simplifyStmt = function
     | e -> e
 
 let rec iterateSimplifyAndInline li =
-    let li =
-        li
-        |> Analyzer.resolve
-        |> Analyzer.markWrites
-        |> Analyzer.maybeInlineVariables
     if not options.noInlining then
+        let li =
+            li
+            |> Analyzer.resolve
+            |> Analyzer.markWrites
         Analyzer.markInlinableFunctions li
-        let mapExpr _ e = e
-        let mapStmt = function
-            | Block b as e -> Analyzer.markInlinableVariables b; e
-            | e -> e
-        mapTopLevel (mapEnv mapExpr mapStmt) li |> ignore
+        Analyzer.markInlinableVariables li
```

Comment:
you shadow `li` in this block, then ignore the value, and use the previous `li` on line 548?

Reply:
Yes. This makes explicit the fact that the inlining passes never change the structure of the tree.
They only mark idents, which is consistent with their name.

Reply:
Then don't assign it to li, this is confusing. You can call ignore.

Reply:
or better, `resolve` and `markWrites` should have type unit